test/iron-fit-behavior.html

```diff
@@ -0,0 +1,302 @@
+        width: 200px;
+      }
+
+      .sized-y {
+        height: 200px;
+      }
+
+      .positioned-left {
+        left: 100px;
+      }
+
+      .positioned-right {
+        right: 100px;
+      }
+
+      .positioned-top {
+        top: 100px;
+      }
+
+      .positioned-bottom {
+        bottom: 100px;
+      }
+
+      .with-max-width {
+        max-width: 500px;
+      }
+
+      .with-max-height {
+        max-height: 500px;
+      }
+
+      .with-margin {
+        margin: 20px;
+      }
+
+    </style>
+
+  </head>
+  <body>
+
+    <test-fixture id="sized-xy">
+      <template>
+        <test-fit auto-fit class="sized-x sized-y">
+          Sized (x/y), auto center/center
+        </test-fit>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="sized-x">
+      <template>
+        <test-fit auto-fit class="sized-x">
+          Sized (x), auto center/center
+        </test-fit>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="positioned-xy">
+      <template>
+        <test-fit auto-fit class="sized-x positioned-left positioned-top">
+          Sized (x/y), positioned/positioned
+        </test-fit>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="inline-positioned-xy">
+      <template>
+        <test-fit auto-fit class="sized-x sized-y" style="right:100px;bottom:100px;">
+          Sized (x/y), positioned/positioned
+        </test-fit>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="sectioned">
+      <template>
+        <test-fit auto-fit class="sized-x">
+          <div>
+            Sized (x), auto center/center with scrolling section
+          </div>
+          <div class="internal"></div>
+        </test-fit>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="constrain-target">
+      <template>
+        <div class="constrain" style="position: fixed; top: 0; left: 0; width: 50vw; height: 50vh; border: 1px solid black;">
+          <test-fit auto-fit class="el">
+            <div>
+              Auto center/center to parent element
+            </div>
+          </test-fit>
+        </div>
+      </template>
+    </test-fixture>
+
+    <template id="ipsum">
+      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+    </template>
+
+    <script>
+
+      function makeScrolling(el) {
+        el.classList.add('scrolling');
+        var template = document.getElementById('ipsum');
+        for (var i = 0; i < 20; i++) {
+          el.appendChild(template.content.cloneNode(true));
+        }
+      }
+
+      suite('manual positioning', function() {
```

Comment:
Do you mean it's not clear why these tests exist? I want to test that if an element is positioned by the user, this code does not try to center it.